Integrations/python/deephaven/learn/__init__.py

```diff
@@ -0,0 +1,226 @@
+    """
+
+    def __init__(self, colName, scatterFunc, type=None):
+        """
+        Initializes an Output object with the given arguments.
+
+        Args:
+            colName         : name of the new column that will store results.
+            scatterFunc     : function that determines how data is taken from a python object into a Deephaven table.
+            type (optional) : desired data type of the new output column.
+        """
+        self.output = _Output_(colName, scatterFunc, type)
+
+    def __str__(self):
+        """
+        Returns the Output object as a string.
+
+        Returns:
+            Output as a string.
+        """
+        return self.output.toString()
+
+
+@_passThrough
+def _validate(inputs, outputs, table):
+    """
+    Ensures that all input columns exist in the table, and that no output column names already exist in the table.
+
+    Args:
+        inputs  : list of Inputs to validate.
+        outputs : list of Outputs to validate.
+        table   : table to check Input and Output columns against.
+
+    Raises:
+        ValueError : if at least one of the Input columns does not exist in the table.
+        ValueError : if at least one of the Output columns already exists in the table.
+    """
+
+    inputColumns = [input.input.getColNames()[i] for input in inputs for i in range(len(input.input.getColNames()))]
+
+    if table.hasColumns(inputColumns):
+
+        if not outputs == None:
+            outputColumns = [output.output.getColName() for output in outputs]
+
+            if not table.hasColumns(outputColumns):
+                return
+
+            else:
+                overlap = set(outputColumns).intersection(table.getMeta().getColumn("Name").getDirect())
+                raise ValueError(f"The columns {overlap} already exist in the table. Please choose Output column names that are not already in the table.")
+
+        return
+
+    else:
+        difference = set(inputColumns).difference(table.getMeta().getColumn("Name").getDirect())
+        raise ValueError(f"Cannot find columns {difference} in the table. Please check your spelling and try again.")
+
+
+@_passThrough
+def _createNonconflictingColumnName(column, table):
+    """
+    Ensures that the given column is not present in the table, and modifies it if necessary until it is not in the table.
+
+    Args:
+        column : column name that should not exist in the table.
+        table  : table to check column against.
+
+    Returns:
+        the column name, modified if necessary so that it is not present in the table.
+    """
+
+    if not table.hasColumns(column):
+        return column
+
+    else:
+        i = 0
+        while table.hasColumns(column):
+            column = column + str(i)
+
+            if not table.hasColumns(column):
+                return column
+            i += 1
+
+
+@_passThrough
+def exec(table=None, model_func=None, inputs=[], outputs=[], batch_size = None):
```

Comment:
Function must be renamed.  `exec` is a built-in python function name.  I already made this comment, and it was not fixed, but the comment was cleared.

Reply:
Yep, this is one of the things I caught after sending for last review. I've renamed it learn for now, but I don't think that's great.